main.py

```diff
@@ -0,0 +1,107 @@
+from modules.dna_rna_tools.py import (
+    is_nucleic_acid, reverse, transcribe, complement, reverse_complement
+    )
+from modules.fastq_tools.py import count_gc, count_quality
+
+
+def run_dna_rna_tools(*args: str) -> "bool | str":
+    """
+    Takes 1 or more sequences, checks if is it nucleic acid and
+    does one of the tools: is_nucleic_acid, reverse, transcribe,
+    complement, reverse_complement
+
+    Arguments:
+    sequences: list[str]
+    tool: str
+
+    Returns bool/str
+    Raises exception if:
+        Number of argumets is less then 2
+        Sequence is not an nucleic acid
+        Tool is unknown
+    """
+    # Checks if there are minimum 2 arguments
+    if len(args) < 2:
+        raise ValueError(
+            "Введите хотя бы одну последовательность и одну процедуру")
+
+    sequences = args[:-1]
+    tool = args[-1]
+
+    # Tools dictionary
+    tools = {
+        "is_nucleic_acid": is_nucleic_acid,
+        "transcribe": transcribe,
+        "reverse": reverse,
+        "complement": complement,
+        "reverse_complement": reverse_complement,
+    }
+
+    # Checks if the tool exists
+    if tool not in tools:
+        raise ValueError("Неизвестная процедура")
+
+    #  Checks if the sequence is nucleic acid
+    for seq in sequences:
+        if tool != "is_nucleic_acid" and not is_nucleic_acid(seq):
+            raise ValueError("Введена некорректная последовательность")
```

Comment:
Огонь, что проверка тут!
Единственное что, стоит поменять порядок проверок, иначе даже если вызывали `is_nucleic_acid` - она пройдётся по всем последовательностям сначала.
```suggestion
if tool != "is_nucleic_acid":
    for seq in sequences:
        if not is_nucleic_acid(seq):
            raise ValueError("Введена некорректная последовательность")
```

Да, так длиннее, но иногда длиннее это лучше.